### PR TITLE
Pictshare only cache image content types. Fixes #709

### LINKED ISSUE #709: Pictshare wrongly caching html pages as well.
@AndreVallestero Okay I shouldn't have merged your PR. I didn't test posting a twitter link, which unfortunately it looks like pictshare will cache: https://communism.lemmy.ml/post/4189

There probably needs to be some type of filetype checking.

### PR DESCRIPTION
**Pull request by dessalines**

### CODE REVIEW

File: server/src/lib.rs [hunk 1]
```diff
@@ -73,6 +73,16 @@ pub fn is_email_regex(test: &str) -> bool {
   EMAIL_REGEX.is_match(test)
 }
 
+pub fn is_image_content_type(test: &str) -> bool {
+  match isahc::get(test) {
+    Ok(res) => match res.headers().get("Content-Type") {
+      Some(header) => header.to_str().unwrap_or("not_an_img").contains("image"),
```
Comment (Nutomic) [inline server/src/lib.rs:79]: It would be better to check for `.startsWith("image/")`. Or even add a whitelist of image mimetypes.

Comment (dessalines) [reply server/src/lib.rs:79]: I tried to find a startsWith, but couldn't find one for rust strings. It just seemed simpler to use contains.

Comment (dessalines) [reply server/src/lib.rs:79]: I think it also might make sense to change this from a bool to just an Ok, Error.

Comment (dessalines) [reply server/src/lib.rs:79]: I just had to merge this quickly and deploy bc every link upload was caching the entire html page in pictshare.

Comment (dessalines) [reply server/src/lib.rs:79]: nm I found the startswith. I'm gonna fix this now.

**Pull request merged**